ta4j-core/src/main/java/org/ta4j/core/TradingRecord.java

```diff
@@ -147,6 +148,11 @@ default Position getLastPosition() {
         return null;
     }
 
+    /**
+     * @return the last position recorded at the given time of 'index'
+     */
+    Position getLastPositionRegardingIndex(int index);
```

Comment:
rename to

`Position getLastPosition(int index){..}`

The "regarding index" is already implicitly told by its needing parameter list `(int index)`